### PR TITLE
Fixed PUBDEV-3281: ARFF parser parses attached file incorrectly

### PR DESCRIPTION
**Pull request by vpatryshev**
The parser did not skip lines with spaces in ARFF header.
Fixed that; and extracted some functionality for that.
Tests added.
Also added some statistics evaluator, so testing is easier now.
ARFF sample test file is in smalldata; how can I get it on s3?

### CODE REVIEW

**Comment by mklechan** [comment]
You can get the smalldata directory and files by ./gradlew syncSmallData

**Comment by vpatryshev** [comment]
Sure; I won't merge the code that does not pass tests.

File: h2o-core/src/main/java/water/parser/ARFFParser.java [hunk 1]
```diff
@@ -112,11 +121,8 @@ private static int readArffHeader(int offset, ArrayList<String> header, byte[] b
       if (bits[lineStart] == '#') continue; // Ignore      comment lines
       if (bits[lineStart] == '%') continue; // Ignore ARFF comment lines
       if (lineEnd > lineStart) {
-        if (bits[lineStart] == '@' &&
-                (bits[lineStart+1] == 'D' || bits[lineStart+1] =='d' ) &&
-                (bits[lineStart+2] == 'A' || bits[lineStart+2] =='a' ) &&
-                (bits[lineStart+3] == 'T' || bits[lineStart+3] =='t' ) &&
-                (bits[lineStart+4] == 'A' || bits[lineStart+4] =='a' )){
+        if (ArrayUtils.matchesInUpperCase(bits, lineStart, DATA_MARKER)) {
+          BytesStats bs = new BytesStats(bits, lineEnd); // for debugging
```
Comment (mmalohlava) [inline h2o-core/src/main/java/water/parser/ARFFParser.java]: This is useless. i do not see `bs` used at all

Comment (vpatryshev) [inline h2o-core/src/main/java/water/parser/ARFFParser.java]: Debugging, yes. What condition should I use?

**Comment by mmalohlava** [comment]
Ups, @h2o-ops reports:
```
java.lang.AssertionError: expected:<10885> but was:<11137>
	at water.parser.ParserTestARFF.testPUBDEV3281()
```

**Comment by mmalohlava** [comment]
@h2o-ops please test!

**Comment by mmalohlava** [comment]
The test is still failing: water.parser.ParserTestARFF.testPUBDEV3281

**Comment by vpatryshev** [comment]
Yes; working on it; thanks.

Thanks,
-Vlad

File: h2o-core/src/test/java/water/junit/H2OTestRunner.java [hunk 2]
```diff
@@ -50,8 +50,9 @@ public Result run(String[] args) throws Exception {
 
   public static void main(String[] args) throws Exception {
     H2OTestRunner testRunner = new H2OTestRunner();
-    Result result = null;
-    result = testRunner.run(args);
+    String doonly = System.getenv("DOONLY");
```
Comment (mmalohlava) [inline h2o-core/src/test/java/water/junit/H2OTestRunner.java]: Please remove it from this PR.

File: h2o-core/src/main/java/water/fvec/FileVec.java [hunk 3]
```diff
@@ -194,7 +194,7 @@ public static int calcOptimalChunkSize(long totalSize, int numCols, long maxLine
     }
     else {
       // New Heuristic
-      int minNumberRows = 10; // need at least 10 rows (lines) per chunk (core)
+      int minNumberRows = 200; // need at least 200 rows (lines) per chunk (core) - that's needed for files with headers
```
Comment (mmalohlava) [inline h2o-core/src/main/java/water/fvec/FileVec.java]: Why you cannot fetch the next chunk in parser/guesser to read header which is spread over multiple chunk.

Comment (vpatryshev) [inline h2o-core/src/main/java/water/fvec/FileVec.java]: I thought that can. But there are cases where chunks go in parallel, like in ParserTestARFF.testNumSplit() - there seems to be no way to figure out when chunks number 0 come in parallel.

**Comment by vpatryshev** [comment]
@h2o-ops please test!

**Comment by mmalohlava** [comment]
@h2o-ops please test!

File: h2o-core/src/main/java/water/fvec/FileVec.java (hunk 3 above)
Comment (mmalohlava) [inline h2o-core/src/main/java/water/fvec/FileVec.java]: Ok, let's keep it as it is for now, but we need to remove that 200^ limit and do the right thing.

Comment (mmalohlava) [inline h2o-core/src/main/java/water/fvec/FileVec.java]: This is follow up JIRA: https://0xdata.atlassian.net/browse/PUBDEV-4223

**Pull request merged**